ArchiveScraping/Segment.py

```diff
@@ -32,13 +32,19 @@ def snippets(self):
 
     def speakerLineGenerator(self):
         '''
-        Skeleton for new method.
-        TODO: Implement function.
-        Should yield the current speaker and the line they said
+        Yields the speaker and line said.
+        If no speaker in transcript return "unknown"
+        TODO be smarter and maybe use NLP to identify the speakers
         '''
         text = self.text()
-        # for speaker in text:
-        #     yield speaker_name, speaker_line
+        
+        for t in text.split('>>'):
+            # each line is seperated by a '>>' string
```

Comment:
I don't think this is true for all segments. I'm getting lots of unknowns and a few mid-identified names. I think the best method might be to use regex to look for a group of letters and a colon. Something like '[a-zA-Z]+:'

Reply:
> I don't think this is true for all segments. I'm getting lots of unknowns and a few mid-identified names. I think the best method might be to use regex to look for a group of letters and a colon. Something like '[a-zA-Z]+:'

Yeah, I tried doing using that exact regex. From at least the examples I used it didn't work well by getting a lot a matches on non-speakers. I tried a lot of different things. The transcripts are not very consistent in this regard. There may be a lot of 'unknown' s because the transcript doesn't record the speaker. Obviously, this implementation isn't perfect and will need improvements.